src/controllers/room.js

```diff
@@ -1,143 +1,99 @@
-
+/* eslint-disable no-underscore-dangle */
 import model from '../database/models';
 
 // creating a room
-
 const createRoom = async (req, res) => {
   try {
     const hotel = await model.hotel.findOne({
-      where : {
+      where: {
         id: req.body.hotelId
       }
-    })
-    if(hotel){
+    });
+    if (hotel) {
       const room = await model.RoomModel.create(req.body);
-
-      hotel.rooms.push(req.body.roomType + "," + " room id: " + room.id)
-      await model.hotel.update({"rooms":hotel.rooms}, {
-        where: { 
-          id:hotel.hotelId
-         }
-      });
       if (room) {
         return res.status(200).json({ room });
       }
-      
-    }else{
-      return res.json({message: res.__("You attempt to assign a room to the hotel which does not exist ! Room not created.")})
+    } else {
+      return res.status(400).json({ message: res.__('You attempt to assign a room to the hotel which does not exist ! Room not created.') });
     }
-    return res.status(200)
-
+    return res.status(200);
   } catch (error) {
-    return res.status(500).json({error: error.message})
+    return res.status(500).json({ error: error.message });
   }
 };
 
-//Getting all rooms
-
-const getAllRooms = async (req, res) => {
-  try {
-    const rooms = await model.RoomModel.findAll();
-    return res.status(200).json({ rooms });
-  } catch (error) {
-    return res.status(500).send(error.message);
-  }
-};
-
-//Getting  a particular room by id
-
+// Getting  a particular room by id
 const getRoomById = async (req, res) => {
   try {
-    const {roomId} = req.params;
+    const { roomId, hotelId } = req.params;
     const room = await model.RoomModel.findOne({
-      where: { id: roomId }
-      
+      where: { id: roomId, hotelId }
+
     });
-    if (room) {
-      return res.status(200).json({ room });
-    }
-    return res.status(404).json({meassage: res.__('Room with the specified ID does not exists')});
-  } catch (error) {
+    const hotel = await model.hotel.findOne({ where: { id: hotelId } });
+
+    if (!hotel) res.status(404).json({ message: res.__('Hotel with the specified ID does not exists') });
 
-    return res.status(500).send(error.message);
+    if (!room) res.status(404).json({ meassage: res.__('Room with the specified ID does not exists') });
+
+    return res.status(200).json({ room });
+  } catch (error) {
+    return res.status(500).send({ message: error.message });
```

Comment:
I think at some point we will have to revise all error messages to be known beforehand and userfriendly

Reply:
I resolve it right now